### PR TITLE
🔧 Modernise build tools

### PR DESCRIPTION
**Pull request by chrisjsewell**
Use flit and move all the tool configuration into pyproject.toml

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,80 @@
...
+    "pylint~=2.12.2",
+    "mypy==0.930",
+]
+
+[project.scripts]
+aiida-mock-code = "aiida_testing.mock_code._cli:run"
+
+[project.entry-points."pytest11"]
+aiida_mock_code = "aiida_testing.mock_code"
+aiida_export_cache = "aiida_testing.export_cache"
```
Comment (chrisjsewell) [inline pyproject.toml:77]: FYI you don't need to load via `pytest_plugins`, if you include this entry point 😉

Comment (ltalirz) [reply pyproject.toml:77]: Oh, nice feature!

Also a bit "dangerous" - I guess any package can now drop any fixture into your pytest setup ;-)

**Review by ltalirz — APPROVED** [review]
thanks a lot @chrisjsewell , all good!

File: pyproject.toml (hunk 1 above)
Comment (chrisjsewell) [reply pyproject.toml:77]: well you likely wouldn't be installing this package if you didn't want to use the fixtures 😄

**Pull request merged**